kolibri/plugins/user/assets/src/actions.js

```diff
@@ -9,50 +9,83 @@
 const constants = require('./state/constants');
 const PageNames = constants.PageNames;
 // const samePageCheckGenerator = require('kolibri.coreVue.vuex.actions').samePageCheckGenerator;
-
+const SignUpResource = require('kolibri').resources.SignUpResource;
+const coreActions = require('kolibri.coreVue.vuex.actions');
+const coreGetters = require('kolibri.coreVue.vuex.getters');
 
 // ================================
 // USER ACTIONS
 
 
+function redirectToHome() {
+  window.location = '/';
+}
+
 /**
  * Actions
  *
  * These methods are used to update client-side state
  */
 
 function showSignIn(store) {
+  console.log(coreGetters.isUserLoggedIn(store.state));
+  if (coreGetters.isUserLoggedIn(store.state)) {
+    redirectToHome();
+    return;
+  }
   store.dispatch('SET_PAGE_NAME', PageNames.SIGN_IN);
   store.dispatch('SET_PAGE_STATE', {});
   store.dispatch('CORE_SET_PAGE_LOADING', false);
   store.dispatch('CORE_SET_ERROR', null);
   store.dispatch('CORE_SET_TITLE', 'User Sign In');
 }
 function showSignUp(store) {
+  if (coreGetters.isUserLoggedIn(store.state)) {
+    redirectToHome();
+    return;
+  }
   store.dispatch('SET_PAGE_NAME', PageNames.SIGN_UP);
   store.dispatch('SET_PAGE_STATE', {});
   store.dispatch('CORE_SET_PAGE_LOADING', false);
   store.dispatch('CORE_SET_ERROR', null);
   store.dispatch('CORE_SET_TITLE', 'User Sign Up');
 }
 function showProfile(store) {
+  if (!coreGetters.isUserLoggedIn(store.state)) {
+    redirectToHome();
+    return;
+  }
   store.dispatch('SET_PAGE_NAME', PageNames.PROFILE);
   store.dispatch('SET_PAGE_STATE', {});
   store.dispatch('CORE_SET_PAGE_LOADING', false);
   store.dispatch('CORE_SET_ERROR', null);
   store.dispatch('CORE_SET_TITLE', 'User Profile');
 }
-function showScratchpad(store) {
-  store.dispatch('SET_PAGE_NAME', PageNames.SCRATCHPAD);
-  store.dispatch('SET_PAGE_STATE', {});
-  store.dispatch('CORE_SET_PAGE_LOADING', false);
-  store.dispatch('CORE_SET_ERROR', null);
-  store.dispatch('CORE_SET_TITLE', 'User Scratchpad');
+
+
+function signUp(store, signUpCreds) {
+  if (coreGetters.isUserLoggedIn(store.state)) {
+    redirectToHome();
+    return;
+  }
+  const signUpModel = SignUpResource.createModel(signUpCreds);
+  const signUpPromise = signUpModel.save(signUpCreds);
+  signUpPromise.then(() => {
+    store.dispatch('CORE_SET_SIGN_UP_ERROR', null);
```

Comment:
@DXCanas Confirmed with Khang that this will be an overlay, not a page. Mockups will be updated by tomorrow.